web/src/app/modules/data/modules/data-service/services/specmate-data.service.ts

```diff
@@ -455,4 +461,36 @@ export class SpecmateDataService {
             await this.readElement(parent, false);
         }
     }
+
+    public async saveModelImage(element: IContainer) {
+        if (Type.is(element, CEGModel) || Type.is(element, Process)) {
+            let model = element as CEGModel | Process;
+            let svg: SVGSVGElement = document.getElementById('mxGraphContainer').getElementsByTagName('svg')[0];
+            let minWidth = svg.style.minWidth;
+            let minHeight = svg.style.minHeight;
+            svg.style.width = '0px';
+            svg.style.height = '0px';
+
+            let maxWidth = parseInt(minWidth.substring(0, minWidth.length - 2));
+            let maxHeight = parseInt(minHeight.substring(0, minHeight.length - 2));
+            let factor = 0.0 + 200 / maxWidth / 1.25;
+            maxHeight = Math.min(maxWidth, maxHeight);
+            let uri: string = await saveAsPng.svgAsPngUri(svg, { 'scale': factor, 'height': maxHeight, 'encoderOptions': 1.0 });
```

Comment:
what is happening here? Is the model saved to disk? is it saved to some remote service? (the latter would not be could)
what is in the ur?